src/vtep6.c

```diff
@@ -0,0 +1,211 @@
+} __attribute__((__packed__));
+
+static void  multicast6_internal(struct vtep_state *state,
+				struct vtep_port *port,
+				union pg_ipv6_addr *multicast_ip,
+				enum operation action,
+				struct pg_error **errp);
+
+
+static void multicast6_subscribe(struct vtep_state *state,
+				struct vtep_port *port,
+				union pg_ipv6_addr multicast_ip,
+				struct pg_error **errp)
+{
+	multicast6_internal(state, port, &multicast_ip, MLD_SUBSCRIBE,
+			    errp);
+}
+
+static void multicast6_unsubscribe(struct vtep_state *state,
+				  struct vtep_port *port,
+				  union pg_ipv6_addr multicast_ip,
+				  struct pg_error **errp)
+{
+	multicast6_internal(state, port, &multicast_ip, MLD_UNSUBSCRIBE,
+			    errp);
+}
+
+
+static void multicast6_internal(struct vtep_state *state,
+				struct vtep_port *port,
+				union pg_ipv6_addr *multicast_ip,
+				enum operation action,
+				struct pg_error **errp)
+{
+	struct rte_mempool *mp = pg_get_mempool();
+	struct rte_mbuf *pkt[1];
+	struct multicast_pkt *hdr;
+	struct ether_addr dst = pg_multicast_get_dst_addr(*multicast_ip);
+
+	if (!pg_is_multicast_ip(multicast_ip)) {
+		*errp = pg_error_new("invalid multicast address %s",
+				     pg_ipv6_to_str(multicast_ip->word8));
+		return;
+	}
+
+	pkt[0] = rte_pktmbuf_alloc(mp);
+	if (!pkt[0]) {
+		*errp = pg_error_new("packet allocation failed");
+		return;
+	}
+	hdr = (void *)rte_pktmbuf_append(pkt[0],
+					 sizeof(struct multicast_pkt));
+
+	ether_addr_copy(&state->mac, &hdr->ethernet.s_addr);
+	ether_addr_copy(&dst, &hdr->ethernet.d_addr);
+	hdr->ethernet.ether_type = rte_cpu_to_be_16(ETHER_TYPE_IPv6);
+
+	union pg_ipv6_vtc vtc = {.version = 6, .traffic_class = 0,
+				 .flow_label = 0};
+	hdr->ipv6.vtc_flow = vtc.vtc_flow;
+	hdr->ipv6.payload_len = rte_cpu_to_be_16(sizeof(struct multicast_pkt) -
+		sizeof(struct ether_hdr));
+	#define ICMPV6_PROTOCOL_NUMBER 58
+	hdr->ipv6.proto = ICMPV6_PROTOCOL_NUMBER;
+	hdr->ipv6.hop_limits = 0xff;
+
+	/* the header checksum computation is to be offloaded in the NIC */
+	pg_ip_copy(hdr->ipv6.src_addr, &state->ip);
+	switch (action) {
+	case MLD_SUBSCRIBE:
+		pg_ip_from_str(hdr->ipv6.dst_addr, "ff02::2");
+		break;
+	case MLD_UNSUBSCRIBE:
+		pg_ip_from_str(hdr->ipv6.dst_addr, "ff02::1");
+		break;
+	default:
+		*errp = pg_error_new("action not handle");
+		goto clear;
+	}
+	hdr->mld.type = action;
+	hdr->mld.code = 0;
+	hdr->mld.cksum = 0;
+	hdr->mld.max_response_time = 0;
+	hdr->mld.reserved = 0;
+	pg_ip_copy(&hdr->mld.multicast_addr, multicast_ip);
+	pg_brick_side_forward(&state->brick.sides[state->output],
+			      pg_flip_side(state->output),
+			      pkt, pg_mask_firsts(1), errp);
+
+clear:
+	rte_pktmbuf_free(pkt[0]);
+}
+
+static struct pg_brick_ops vtep_ops = {
+	.name		= BRICK_NAME,
+	.state_size	= sizeof(struct vtep_state),
+
+	.init		= vtep_init,
+	.destroy	= vtep_destroy,
+
+	.unlink		= pg_brick_generic_unlink,
+
+	.unlink_notify  = vtep_unlink_notify,
+};
+
+pg_brick_register(vtep, &vtep_ops);
```

Comment:
Don't forget the [checklist](https://github.com/outscale/packetgraph/blob/master/doc/contrib.md#checklist), I think you need to complete `README.md` ;)